### PR TITLE
throw unimplemented errors

### LINKED ISSUE #323: throw 'unimplemented' errors for API queries that aren't and won't be implemented within MMR scope
currently, when you graphql request for `Agent.relationships`, for example, you just get `null`. Preferable would be to get an error back, informing the developer the endpoint isn't implemented yet. Otherwise they won't know if they're just getting a normal null field or an unimplemented one.

## Query
This would all happen here: https://github.com/h-REA/hREA/tree/sprout/modules/vf-graphql-holochain/queries
- [x] requests
- [x] offers
- [x] agentRelationship
- [x] agentRelationships
- [x] agentRelationshipRole
- [x] agentRelationshipRoles

## Mutation
- [x] createAgentRelationship
- [x] updateAgentRelationship
- [x] deleteAgentRelationship
- [x] createAgentRelationshipRole
- [x] updateAgentRelationshipRole
- [x] deleteAgentRelationshipRole

## Resolvers
The rest would all happen here: https://github.com/h-REA/hREA/tree/sprout/modules/vf-graphql-holochain/resolvers

### Agent
- [x] Agent.relationships
- [x] Agent.relationshipsAsSubject
- [x] Agent.relationshipsAsObject
- [x] Agent.roles
- [x] Agent.commitments
- [x] Agent.intents
- [x] Agent.economicEvents
- [x] Agent.commitmentsInScope
- [x] Agent.economicEventsInScope
- [x] Agent.plans
- [x] Agent.processes
- [x] Agent.proposals
- [x] Agent.proposalsInScope
- [x] Agent.proposalsTo

### Agreement
- [x] unplannedEconomicEvents

### Commitment
- [x] stage
- [x] resourceInventoriedAs

### Fulfillment
Fine

### Intent
- [x] resourceInventoriedAs 

### Measure
Fine

### Economic Event
- [x] triggeredBy
- [x] triggers
- [x] previous
- [x] next
- [x] track
- [x] trace

### Economic Resource
- [x] economicEventsInOutFrom
- [x] economicEventsTo
- [x] previous
- [x] next
- [x] track
- [x] trace
- [x] commitments
- [x] intents

### Plan
- [x] involvedAgents

### Process
- [x] nextProcesses
- [x] previousProcesses
- [x] involvedAgents
- [x] unplannedInputs
- [x] unplannedOutputs
- [x] previous
- [x] next

### Process Specification
This file is missing completely, and should be added. The resolvers can remain unimplemented, but the following should exist:
- [x] commitmentsRequiringStage
- [x] resourcesCurrentlyAtStage
- [x] conformingProcesses

### Proposal
- [x] primaryIntents
- [x] reciprocalIntents

### Proposed Intent
Fine

### ProposedTo
Fine

### Resource Specification
- [x] defaultUnitOfResource (is tracked by issue #155)
- [x] commitments
- [x] intents
- [x] economicEvents

### Satisfaction
Fine (except that I am noting that the resolvers should be made conditional on VfModules enabled

### PR DESCRIPTION
**Pull request by weswalla**
closes #323

### CODE REVIEW

**Comment by Connoropolous** [comment]
Nice. Which ones have we vetted so far between you and me?

**Comment by weswalla** [comment]
@Connoropolous covered all the ones outlined in #323

File: modules/vf-graphql-holochain/resolvers/agent.ts [hunk 1]
```diff
@@ -89,18 +115,29 @@ export default (enabledVFModules: VfModule[] = DEFAULT_VF_MODULES, dnaConfig: DN
...
+        // return proposals
+        throw new Error('resolver unimplemented')
+      },
+      proposalsInScope: async (record: Agent): Promise<ProposalConnection> => {
+        throw new Error('resolver unimplemented')
+      },
+      proposalsTo: async (record: Agent): Promise<ProposalConnection> => {
+        throw new Error('resolver unimplemented')
+      },
+      proposalsToWithin: async (record: Agent): Promise<ProposalConnection> => {
```
Comment (Connoropolous) [inline modules/vf-graphql-holochain/resolvers/agent.ts]: this one doesn't exist and caused an error during my local test run. remove

**Review by Connoropolous — CHANGES_REQUESTED** [review]

**Comment by Connoropolous** [comment]
@weswalla check out the issue again, I've updated it with what I hope is the full list

File: modules/vf-graphql-holochain/resolvers/commitment.ts [hunk 2]
```diff
@@ -111,5 +120,10 @@ export default (enabledVFModules: VfModule[] = DEFAULT_VF_MODULES, dnaConfig: DN
         return readPlan(record, { id: record.plannedWithin })
       },
     } : {}),
+    (hasObservation ? {
+      resourceInventoriedAs: async (record: Commitment): Promise<EconomicResource> => {
```
Comment (Connoropolous) [inline modules/vf-graphql-holochain/resolvers/commitment.ts:124]: this type `Commitment` isn't right I think

**Review by Connoropolous — APPROVED** [review]

**Pull request merged**